pkg/controller/elasticsearch/nodespec/podspec.go

```diff
@@ -134,7 +134,7 @@ func BuildPodTemplateSpec(
 		WithContainersSecurityContext(securitycontext.For(ver, enableReadOnlyRootFilesystem)).
 		WithPreStopHook(*NewPreStopHook())
 
-	builder, err = stackmon.WithMonitoring(ctx, client, builder, es)
+	builder, err = stackmon.WithMonitoring(ctx, client, builder, es, enableReadOnlyRootFilesystem)
```

Comment:
nit: The way we decide to include those empty volumes is not the same between Elasticsearch and Logstash. For Logstash we try to infer that decision using `isReadOnlyRootFilesystem(...)` on the pod spec. While for Elasticsearch we are only using our custom logic, however `enableReadOnlyRootFilesystem` does not mean that the file system is not going to be "read-only". Maybe this has been enforced by the user in the ES pod template.
Should we be consistent and use `isReadOnlyRootFilesystem(...)` in all the controllers? (or just add the empty volumes by default as suggested in my previous comment)

Reply:
> or just add the empty volumes by default as suggested in my previous comment

+1 to that proposal. It removes all the conditional logic and makes the code simpler and it is never a good idea to write into a containers filesystem no matter what the security settings are.